### ISSUE TITLE
Error loading key from attachment, "Attachment does not exist" and NullReferenceException

### ISSUE THREAD

**Issue by nicoell** [dlech/KeeAgent#325]
## Up front
My conclusion while investigating this issue is that "something" corrupted a specific entry in my KeyPass Database. Apparently, the file-content of the attachments (private and public key) of this entry got replaced with the contents of another entry in the same group. What followed is that KeeAgent seemingly behaved odd, as detailed below.

While it would be nice to find the core issue of what corrupted my entry, I don't know how I could track that down. I see many potential reasons, e.g. old KeePass version on another device, outdated plugins, synchronization with cloud. 

So, I don't want to blame KeeAgent here, I rather thought telling this "story" could still be interesting. Even if it's just the insight that there is a case where GetSshKey can be called with getPrivateKeyStream returning null.

## Manifestation of the issue
This error started to appear recently without any prior changes to entries with Keys. Since then it appears every time I load my KeyAgent database.
![image](https://user-images.githubusercontent.com/4254495/143414299-266fa46b-9208-4b2c-9090-6c517f008444.png)

Looking at the code, the function seems to expect the PrivateKeyStream to never be null, but it is:
https://github.com/dlech/KeeAgent/blob/3caecd7ff40a3a282e51a61f4e8fdbbf7f093be2/KeeAgent/ExtensionMethods.cs#L206-L222

## PrivateKey is null because "Attachment does not exist"

First, after pressing OK on above error, a dialogue pops up:
![image](https://user-images.githubusercontent.com/4254495/143414470-6de13fe3-7884-4d70-8b3b-61bb03a67876.png)
No further error appears on loading additional keys, but according to the Key Info Comment of entries with Keys, all of them fail to load with the reason:
_\<Error loading key from attachment\>_

For all my entries, the Private Key File Location is set to an attachment file (filenames pixelated in image below):
I can open and view all File attachments without any problems:
![image](https://user-images.githubusercontent.com/4254495/143417136-08bc5ece-1330-4347-99e2-02322e96fb2f.png)

But, on pressing OK during "Edit Entry", a popup tells me "Attachment does not exist".
![image](https://user-images.githubusercontent.com/4254495/143416061-5c0ad2f4-d1cd-4f92-a3c0-7b57c7ece91f.png)

This dialogue also prevents me from changing anything from the entry.

### A new error appears
After disabling and enabling the flag for KeeAgent to use this Entry, the Key Info Comment still says _\<Error loading key from attachment\>_, but I can change the entry without the "Attachment does not exist" popup.

But, on reopening the database, I now get a different error dialogue:
![image](https://user-images.githubusercontent.com/4254495/143419223-84823e83-7ee4-4987-9344-8b7d0a3a4385.png)

### The culprit of all this
After taking a closer look at the private key in the attachments, I noticed that it is not the correct private key. It is exactly the same private key as another entry within the same Group: My GitHub private key has become my GitLab private key. Same for the public key that I also saved in the attachments.
I don't know how or when this happened and there is no valid previous version in the Entry History that I could restore, just one very old and outdated one without keys.

**Comment by dlech** [comment]
Is it possible you have two entries with the same UUID?

![image](https://user-images.githubusercontent.com/963645/151644747-d34e3bab-ae7c-49f6-9d7b-7294b01fd872.png)

**Comment by nicoell** [comment]
No, all UUIDs are unique.

I have three more entries with scrambled attachments. They got randomized... Private Keys became KeeAgent settings, Public keys of one entry became the private key of another entries. 

I did some more research and my best guess is that this was caused by [KeyPassDX](https://github.com/Kunzisoft/KeePassDX) on Android. There were related issues reported:
https://github.com/Kunzisoft/KeePassDX/issues/726
